operators/src/processing/reprojection.rs

```diff
@@ -0,0 +1,399 @@
+use std::marker::PhantomData;
+
+use futures::TryStreamExt;
+use geoengine_datatypes::{
+    operations::reproject::{CoordinateProjection, CoordinateProjector, Reproject},
+    spatial_reference::SpatialReference,
+};
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+use crate::{
+    engine::{
+        ExecutionContext, InitializedOperator, InitializedOperatorImpl, InitializedVectorOperator,
+        Operator, QueryContext, QueryRectangle, TypedVectorQueryProcessor, VectorOperator,
+        VectorQueryProcessor, VectorResultDescriptor,
+    },
+    error,
+    util::Result,
+};
+
+use super::map_query::MapQueryProcessor;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ReprojectionParams {
+    pub target_spatial_reference: SpatialReference,
+}
+
+pub type Reprojection = Operator<ReprojectionParams>;
+pub type InitializedReprojection =
+    InitializedOperatorImpl<(), VectorResultDescriptor, (SpatialReference, SpatialReference)>;
```

Comment:
You have to be careful about using the unit type. I have also done this, but you can't say

`type a =
    InitializedOperatorImpl<(), VectorResultDescriptor, ()>;`

and then

`type b =
    InitializedOperatorImpl<(), VectorResultDescriptor, ()>;`

because it is the same type.

I started using empty structs for the impls when I saw this failing on a second type.

Reply:
is there an exampe for this?

Reply:
This is the counter example:
https://github.com/geo-engine/geoengine/blob/master/operators/src/processing/point_in_polygon.rs

You can't say `pub type PointInPolygonFilterB = Operator<()>;` or `pub type InitializedPointInPolygonFilterB = InitializedOperatorImpl<(), VectorResultDescriptor, ()>;`.

Reply:
bb87c88